File: backend/src/main/java/ecommerce/dto/Purchase.java
package ecommerce.dto;

import ecommerce.entity.*;
import lombok.Data;

import java.util.Set;

@Data
public class Purchase {
    private Customer customer;
    private Address shippingAddress;
    private Address billingAddress;
    private Order order;
    private ShippingMethod shippingMethod;
    private Set<OrderItem> orderItems;
}

File: frontend/src/app/common/purchase.ts
import { ShippingMethod } from './shipping-method';
import { OrderItem } from './order-item';
import { Order } from './order';
import { Address } from './address';
import { Customer } from './customer';
export class Purchase {
    customer: Customer;
    shippingAddress: Address;
    billingAddress: Address;
    shippingMethod: ShippingMethod;
    order: Order;
    orderItems: OrderItem[];
}
